platform/features/conductor-v2/conductor/src/ui/ConductorTOIController.js

```diff
@@ -0,0 +1,110 @@
+
+define(
+    ["zepto"],
+    function ($) {
+
+        /**
+         * The mct-conductor-axis renders a horizontal axis with regular
+         * labelled 'ticks'. It requires 'start' and 'end' integer values to
+         * be specified as attributes.
+         */
+        function ConductorTOIController($scope, openmct, conductorViewService) {
+            this.conductor = openmct.conductor;
+            this.conductorViewService = conductorViewService;
+
+            //Bind all class functions to 'this'
+            Object.keys(ConductorTOIController.prototype).filter(function (key) {
+                return typeof ConductorTOIController.prototype[key] === 'function';
+            }).forEach(function (key) {
+                this[key] = ConductorTOIController.prototype[key].bind(this);
+            }.bind(this));
+
+            this.conductor.on('timeOfInterest', this.changeTimeOfInterest);
+            this.conductorViewService.on('zoom', this.setOffsetFromBounds);
+            this.conductorViewService.on('pan', this.setOffsetFromBounds);
+
+            var timeOfInterest = this.conductor.timeOfInterest();
+            if (timeOfInterest) {
+                this.changeTimeOfInterest(timeOfInterest);
+            }
+
+            $scope.$on('$destroy', this.destroy);
+
+        }
+
+        ConductorTOIController.prototype.destroy = function () {
+            this.conductor.off('timeOfInterest', this.changeTimeOfInterest);
+            this.conductorViewService.off('zoom', this.setOffsetFromBounds);
+            this.conductorViewService.off('pan', this.setOffsetFromBounds);
+        };
+
+        /**
+         * Given some bounds, set horizontal position of TOI indicator based
+         * on current conductor TOI value. Bounds are provided so that
+         * ephemeral bounds from zoom and pan events can be used as well
+         * as current conductor bounds, allowing TOI to be updated in
+         * realtime during scroll and zoom.
+         * @param {TimeConductorBounds} bounds
+         */
+        ConductorTOIController.prototype.setOffsetFromBounds = function (bounds) {
+            var toi = this.conductor.timeOfInterest();
+            if (toi !== undefined) {
+                var offset = toi - bounds.start;
+                var duration = bounds.end - bounds.start;
+                this.left = offset / duration * 100;
+                this.pinned = true;
+            } else {
+                this.left = 0;
+                this.pinned = false;
+            }
+        };
+
+        ConductorTOIController.prototype.changeTimeOfInterest = function () {
+            var bounds = this.conductor.bounds();
+            if (bounds) {
+                this.setOffsetFromBounds(bounds);
+            }
+        };
+
+        /**
+         * Set time of interest
+         * @param e The angular $event object
+         */
+        ConductorTOIController.prototype.click = function (e) {
```

Comment:
Would it be clearer to name this function based on what it does rather than the event it is following?